src/PrestaShopBundle/Resources/views/Admin/Module/Includes/action_menu.html.twig

```diff
@@ -23,47 +23,73 @@
-                      <button type="submit" class="dropdown-item module_action_menu_{{ module_action }}" data-confirm_modal="module-modal-confirm-{{ name }}-{{ module_action }}">
-                        {{module_action|capitalize|replace({'_': " "})|trans({}, 'Admin.Actions')}}
-                      </button>
-                    </form>
-                  </li>
-              {% endif %}
-            {% endfor %}
+      <div class="btn-group form-action-button-container">
+          <form class="btn-group form-action-button import-module-from-popin-form" method="post" action="{{ urls[url_active] }}">
+              <button type="submit"
+                      class="btn btn-primary btn-reverse btn-outline-primary light-button module_action_menu_{{ url_active }}"
+                      data-confirm_modal="module-modal-confirm-{{ name }}-{{ url_active }}">
+                  {{ url_active|capitalize|replace({'_': " "})|trans({}, 'Admin.Actions') }}
+              </button>
+          </form>
+
+          <div id="loading_message" style="display: none;">
+              <p class=module-import-processing-main-text>{{ 'In progress...'|trans({}, 'Admin.Modules.Notification') }}</p>
           </div>
-    {% endif %}
-  </div>
+
+          {% if (urls|length > 1) %}
+              <input type="hidden" class="btn">
+              <button type="button" class="btn btn-outline-primary dropdown-toggle dropdown-toggle-split"
+                      data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+                  <span class="caret"></span>
+                  <span class="sr-only">{{ 'Toggle Dropdown'|trans({}, 'Admin.Modules.Feature') }}</span>
+              </button>
+              <div class="dropdown-menu">
+                  {% for module_action, module_url in urls %}
+                      {% if module_action != url_active %}
+                          <li>
+                              <form class="import-module-from-popin-form" method="post" action="{{ module_url }}">
+                                  <button type="submit" class="dropdown-item module_action_menu_{{ module_action }}"
+                                          data-confirm_modal="module-modal-confirm-{{ name }}-{{ module_action }}">
+                                      {{ module_action|capitalize|replace({'_': " "})|trans({}, 'Admin.Actions') }}
+                                  </button>
+                              </form>
+                          </li>
+                      {% endif %}
+                  {% endfor %}
+              </div>
+          {% endif %}
+      </div>
   {% endif %}
+
+<script>
+    $(document).ready(function () {
+        $('#import-module-from-popin-form').on('submit', function (e) {
+            e.preventDefault();
+            $(this).after( "<p class=module-import-processing-main-text>{{ 'Installing module...'|trans({}, 'Admin.Modules.Notification') }}</p>" );
```

Comment:
Play with show & hide instead injecting class everytime. Otherwise in case of failure you will see Installing module twice.